src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRunner.cs

```diff
@@ -209,6 +218,19 @@ public static async Task<IReadOnlyList<RestoreSummaryRequest>> GetRequests(Resto
             return requests;
         }
 
+        /// <summary>
+        /// If the request does not have a ParentId add one.
+        /// This will enable event telemetry correlation.
+        /// </summary>
+        /// <param name="request">The request</param>
+        private static void AddParentIdToRequest(RestoreSummaryRequest request)
```

Comment:
When is this generated/not generated today? 

Can we just generate the parent id at the beginning when creating this instead of populating it later.
Populating missing data can be hard to follow sometimes imho.

Reply:
I think that this is a good suggestion. The parent Id is read from the RestoreArgs [here](https://github.com/NuGet/NuGet.Client/blob/ac4c233093196057c112c19e05a869c8ea69892a/src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs#L194) and  [here](https://github.com/NuGet/NuGet.Client/blob/ac4c233093196057c112c19e05a869c8ea69892a/src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs#L240).
I updated to set a parentId on the creation of the RestoreArgs. Unfortunately the  RestoreArgs.ParenId is not read only so it could be very easy to be missed to be set. Also there are not any constrains for the field to not be empty. Currently in two random days of data there are 0.05% of devenv projectrestoreinformation events without a parent id. 

If we think that the addition is not necessary I can remove the new setters.